qml/CoverPage.qml

```diff
@@ -45,22 +45,29 @@ CoverBackground {
     ListView {
         id: fileListView
 
-        property real itemHeight: Theme.iconSizeSmall + Theme.paddingSmall*2
-
+        property int itemHeight: height/maxItemCount
+        property int maxItemCount: Math.round(height/(dummy.height + Theme.paddingSmall))
         clip: true
         interactive: false
         model: window.fileListModel
         visible: window.documentItem === null
-        y: Theme.paddingLarge
-        width: parent.width
-        height: 7*itemHeight
+        anchors {
+            fill: parent
+            topMargin: Theme.paddingLarge
+            bottomMargin: Theme.paddingLarge
+        }
 
         delegate: CoverFileItem {
             width: fileListView.width
             height: fileListView.itemHeight
             text: model.fileName
             iconSource: window.mimeToIcon(model.fileMimeType)
         }
+        Label {
```

Comment:
Currently height is explicitly determined by the user of CoverFileItem so I think deriving label height from external item is also ok. That being said I wouldn't normally write layouts like this (implicit height is normally defined by the item itself), but rewriting the layout feels a bit of an overkill too. Adding invisible dummy CoverFileItem would add more loading overhead than adding just a label, I will replace the label with FontMetrics QML element once Qt 5.6 is in use.